houston/ardu/sandbox.py

```diff
@@ -213,86 +221,189 @@ def run(self, commands: Sequence[Command]) -> 'MissionOutcome':
-                                                    state_after,
-                                                    env,
-                                                    config)
-                mission_passed = mission_passed and passed
+                cmd_index = 1 + i
+                state_before = None
+                for j in range(cmd_index - 1, -1, -1):
+                    if j in wp_state:
+                        state_before = wp_state[j][0]
+                        break
+                if cmd_index in wp_state:
+                    state_after = wp_state[cmd_index][0]
+                    time_elapsed = wp_state[cmd_index][1] - wp_state[j][1]
+                    command = commands[i]
+                    directory = 'command{}'.format(cmd_index)
+                    coverage = self.__get_coverage(directory=directory)
+                    m = recorder_filename + '.cov'
+                    with open(m, 'a') as f:
+                        f.write('Command #{}: {}\n'.format(cmd_index, command))
+                        f.write('{}\n'.format(str(coverage)))
+                else:
+                    state_after = state_before
+                    time_elapsed = 0.0
+
+#                determine which spec the system should observe
+#                spec = command.resolve(state_before, env, config)
+#                postcondition = spec.postcondition
+#                passed = postcondition.is_satisfied(command,
+#                                                    state_before,
+#                                                    state_after,
+#                                                    env,
+#                                                    config)
+#                mission_passed = mission_passed and passed
+                passed = True
                 outcome = CommandOutcome(command,
                                          passed,
                                          state_before,
                                          state_after,
                                          time_elapsed)
                 outcomes.append(outcome)
-                state_before = state_after
+#                state_before = state_after
                 mission_time += time_elapsed
 
             return MissionOutcome(mission_passed, outcomes, mission_time)
 
     def update(self, message: Message) -> None:
-        logger.debug("UPDATE")
+        # logger.debug("UPDATE")
         with self.__state_lock:
             self.__state = self.__state.evolve(message,
                                                self.running_time,
                                                self.connection)
-            logger.debug("S: %s", self.state)
+            # logger.debug("S: %s", self.state)
+            if self.recorder:
+                self.recorder.add(self.__state)
+
+    def __get_coverage(self, directory: str) -> "FileLineSet":
+        """
+        Copies gcda files from /tmp/<directory> to /opt/ardupilot
```

Comment:
Feel free to ignore, but couldn't `directory` be an absolute path rather than a path relative to `/tmp`? Fewer assumptions.